File: 507. Perfect Number/solution.js
/**
 * @param {number} num
 * @return {boolean}
 */
var checkPerfectNumber = function(num) {
  let sum = 1;
  for (let i = 2, n = Math.sqrt(num); i < n; i++) {
    if (num % i === 0) sum += i + num / i;
  }
  return num !== 1 && sum === num;
};

File: 507. Perfect Number/solution.go
func checkPerfectNumber(num int) bool {
	if num <= 0 {
		return false
	}

	sum := 0
	for i := 1; i*i <= num; i++ {
		if num%i == 0 {
			sum += i
			if i*i != num {
				sum += num / i
			}
		}
	}

	return sum == num+num
}